pkg/cmd/server/start/start_master.go

```diff
@@ -472,6 +480,10 @@ func (m *Master) Start() error {
 		go func() {
 			controllerPlug.WaitForStart()
 
+			// this does a second leader election, but doing the second leader election will allow us to move out process in
+			// 3.8 if we so choose.
+			go schedulerapp.Run(scheduler)
```

Comment:
Doing this will run a second leader election for the scheduler, but doing it like this will allow us to split into a separate binary/unit file in 3.8 without worrying about ending up with dueling schedulers since the second leader election will gate the separate binary.